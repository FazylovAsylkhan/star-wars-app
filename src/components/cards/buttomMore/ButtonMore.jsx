/* eslint-disable prefer-destructuring */
import React from "react";
import { Link } from "react-router-dom";
import buttonStyle from "../../../scss/button.module.scss";

function ButtonMore({ idPersonalPage }) {
  const buttonClass = `${buttonStyle.button} ${buttonStyle.primary}`;

  return (
    <button className={buttonClass} type="button">
      <Link className={buttonStyle.link} to={`${idPersonalPage}`}>
        More ...
      </Link>
    </button>
  );
}

export default ButtonMore;
