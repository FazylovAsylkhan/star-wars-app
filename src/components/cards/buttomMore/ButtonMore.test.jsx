/* eslint-disable no-undef */
import React from "react";
import { screen } from "@testing-library/react";
import { renderWithRouter } from "../../../tests/testUtils";
import ButtonMore from "./ButtonMore";

describe("ButtonMore component works", () => {
  it("should render ButtonMore component", async () => {
    renderWithRouter(<ButtonMore index={1} />);
    const button = screen.getByText(/More.../i);
    expect(button).toBeInTheDocument();
    expect(button).toMatchSnapshot();
  });
});
