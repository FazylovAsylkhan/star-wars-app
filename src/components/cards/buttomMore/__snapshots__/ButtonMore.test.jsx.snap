// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonMore component works should render ButtonMore component 1`] = `
<a
  class="link"
  href="/undefined"
>
  More ...
</a>
`;
